### PR TITLE
NMS-9783: Include X-RequestedWith header for all  requests

### PR DESCRIPTION
**Pull request by RangerRick**
This is a new version of @mvrueden's PR updated to match the latest JS changes.

* JIRA: http://issues.opennms.org/browse/NMS-9783

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

File: core/web-assets/src/main/assets/js/apps/geomap-js/index.js [hunk 1]
```diff
@@ -90,6 +90,9 @@ const render = function(options) {
         $.ajax({
             method: 'GET',
             url: restEndpoint + '/config',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest'
```
Comment (mvrueden) [inline core/web-assets/src/main/assets/js/apps/geomap-js/index.js]: Nice for also including this (-:

File: core/web-assets/src/main/assets/js/apps/geomap-js/index.js [hunk 2]
```diff
@@ -110,6 +113,9 @@ const render = function(options) {
             method: 'POST',
             url: restEndpoint,
             contentType: 'application/json',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest'
```
Comment (mvrueden) [inline core/web-assets/src/main/assets/js/apps/geomap-js/index.js]: Nice for also including this (-:

**Comment by RangerRick** [comment]
Alright, I updated it and I moved all the raw ajax calls to a separate commit.

**Comment by j-white** [comment]
Should we also update opennms-js to add this header by default?

**Comment by mvrueden** [comment]
Sounds good to me. @RangerRick can you do that?
I will merge this, as the changes to opennms-js need to be done in a separate repository anyways.

**Comment by RangerRick** [comment]
Yeah, that's a good idea.

**Pull request merged**